Orange/data/util.py

```diff
@@ -28,3 +28,27 @@ def scale(values, min=0, max=1):
     if ptp == 0:
         return np.clip(values, min, max)
     return (-minval + values) / ptp * (max - min) + min
+
+
+class ComputeValueWCommon:
+    """This class stuctures compute_value functions that
+    repeat parts of computation for different variables into
+    common and specific parts.
+
+    Attributes:
+        commonfn    Function that performs computation that is common
+                    for multiple variables. All variables that share
+                    computation need to set the same function instance.
+    """
+    commonfn = lambda data: None
+
+    def __call__(self, data, common=None):
+        """Fallback if common parts are not passed."""
+        if common is None:
+            common = self.commonfn(data)
+        return self.specific(common, data)
+
+    def specific(self, common, data):
+        """Given the common computation results, perform variable-specific
+        part of computation and return new variable values."""
+        raise NotImplementedError("specific computation must be implemented")
```

Comment:
No, sure, there's nothing wrong with the line above. I just get to assume every developer who sees a `NotImplementedError` raised in a method of some base or mixin class understands that that is (one of) the method they are intended to override, thoughtfully seeking further information in the method's docstring, most likely. In this view, there is very little that
```py
raise NotImplementedError('you have to override this method')
```
adds compared to just
```py
raise NotImplementedError
```
Again, nothing _wrong_ with the code above. Sorry.